src/jemalloc.c

```diff
@@ -747,8 +747,11 @@ malloc_ncpus(void) {
 #  endif
 #  if defined(JEMALLOC_HAVE_SCHED_SETAFFINITY)
 		sched_getaffinity(0, sizeof(set), &set);
-#  else
+#  elif defined(JEMALLOC_HAVE_PTHREAD_GETAFFINITY_NP)
 		pthread_getaffinity_np(pthread_self(), sizeof(set), &set);
+#  else
+		CPU_ZERO(&set);
```

Comment:
This will cause jemalloc to detect wrong number of CPUs, and likely will not perform well, even defeating the purpose of using jemalloc. Is there no way to do so on hurd?

Reply:
SMP support is still WIP, so it's actually the correct number of CPUs :)
That said, `sysconf(_SC_NPROCESSORS_ONLN)` will properly report 1, so perhaps it's the `#elif defined(CPU_COUNT)` above that should rather be `#elif defined(CPU_COUNT) && (defined(JEMALLOC_HAVE_SCHED_SETAFFINITY) || defined(JEMALLOC_HAVE_PTHREAD_GETAFFINITY_NP)`